### PR TITLE
Add remaining XML-RPC endpoints

### LINKED ISSUE #4: Missing API Endpoints for a lot of things
@nodeg Please use and extend this. **DONE**

- [x] basically all actions except `sync`
- [x] item `copy`
- [x] item `report`
- [x] item `list`
- [x] item `rename`
- [x] profile/system `get-autoinstall`
- [x] everything related to the item `package`
- [x] everything related to the item `mgmtclass`
- [x] everything related to the item `file`
- [x] everything related to the item `menu`
- [x] everything related to the item `image`

### PR DESCRIPTION
**Pull request by SchoolGuy**
I am not sure if this adds all remaining endpoints but it adds the biggest chunk of them.

This should contain tests for most of the endpoints that are being communicated with.

This PR currently doesn't touch the topic of resolved items and exposing this conveniently. I think this would blow up the scope of this PR too much.

Fixes #4

### CODE REVIEW

File: distro.go
Comment (ByteOtter) [inline distro.go:145]: Not sure if these comments are work in progress?

File: distro.go
Comment (ByteOtter) [inline distro.go:151]: WIP comments?

File: event.go
Comment (ByteOtter) [inline event.go:23]: WIP comment?

File: event.go
Comment (ByteOtter) [inline event.go:45]: like above, a bit confusing

File: event.go
Comment (tiltingpenguin) [inline event.go]: Debug print left in the code comment

File: event.go
Comment (tiltingpenguin) [inline event.go]: Grammar error in the docstring and unneeded else statement. Also this function doesn't look done to me since it always returns an EMPTYEVENT

File: item.go
Comment (ByteOtter) [inline item.go:53]: a lot of those here, so I'll place another marker. :)

File: resource.go
Comment (ByteOtter) [inline resource.go]: todo here too? ;P

**Review by ByteOtter — APPROVED** [review]
From my side it looks good - though my Go is not the best so you may want to wait for another review :)

I also marked some todos in regards to documentation for you, take a look if you have time but as long as the functionality works this is fine to merge.

File: resource.go
Comment (SchoolGuy) [reply resource.go]: No, but that can be extended definitely!

File: item.go
Comment (tiltingpenguin) [inline item.go]: Is the absence of an error meant to signal the existence of the items? Also I don't understand what exactly this function is finding if we have no parameters

File: item.go
Comment (tiltingpenguin) [inline item.go]: Unnecessary else. I would cut it for consistency.

File: item.go
Comment (tiltingpenguin) [inline item.go:70]: Do we  want the result, even if the call returns an error? Or is the result nil if it fails? I think explicit error handling here might be good for clarity

File: menu.go
Comment (tiltingpenguin) [inline menu.go]: Unnecessary else

File: mgmtclass.go
Comment (tiltingpenguin) [inline mgmtclass.go]: Wrong docstring

File: mgmtclass.go
Comment (tiltingpenguin) [inline mgmtclass.go]: Unnecessary else

File: mgmtclass.go
Comment (tiltingpenguin) [inline mgmtclass.go]: Use the SaveMgmtClass method instead

File: mgmtclass.go
Comment (tiltingpenguin) [inline mgmtclass.go]: missing annotations

File: resource_test.go
Comment (tiltingpenguin) [inline resource_test.go:1]: Why is this needed?

File: signatures.go
Comment (tiltingpenguin) [inline signatures.go]: Unnecessary else

File: signatures_test.go
Comment (tiltingpenguin) [inline signatures_test.go]: This should be <= 1 according to the error message. Also there is a typo in the message

Same problem in the following tests

File: image.go
Comment (tiltingpenguin) [inline image.go]: Use the SaveImage method instead

File: linuxpackage.go
Comment (tiltingpenguin) [inline linuxpackage.go]: Use the SavePackage method instead

File: linuxpackage.go
Comment (tiltingpenguin) [inline linuxpackage.go]: Use the SavePackage method instead

File: menu.go
Comment (tiltingpenguin) [inline menu.go]: Use SaveMenu method instead

File: menu.go
Comment (tiltingpenguin) [inline menu.go]: Use SaveMenu method instead

File: mgmtclass.go
Comment (tiltingpenguin) [inline mgmtclass.go]: Use SaveMgmtClass method instead

File: file.go
Comment (tiltingpenguin) [inline file.go]: Use SaveFile method instead

File: file.go
Comment (tiltingpenguin) [inline file.go]: Use SaveFile method instead

**Review by tiltingpenguin — CHANGES_REQUESTED** [review]
I think we are missing the tests for item.go

File: event.go
Comment (SchoolGuy) [reply event.go]: That is due to the FIXME.

File: item.go
Comment (SchoolGuy) [reply item.go:70]: A bool cannot be `nil` in Golang. As such it doesn't matter if we add explicit handling since we cannot return `nil`.

File: item.go
Comment (SchoolGuy) [reply item.go]: I added the parameters but the return type is too complex for now to be implemented in this PR.

File: image.go
Comment (tiltingpenguin) [inline image.go]: From the diff to the previously review version this call changed from save_profile to SaveImage. According to the comment as well this should be SaveProfile instead.

**Review by tiltingpenguin — COMMENTED** [review]
I only checked the diff between the last version and this one but apart from one mistake this LGTM

**Review by ByteOtter — APPROVED** [review]
LGTM after you fix tiltingpenguin's comments.

File: image.go
Comment (SchoolGuy) [reply image.go]: Yes but actually both are wrong. This is the method to create an image. As such this is a copy & paste error from me.

**Review by tiltingpenguin — APPROVED** [review]
LGTM

**Pull request merged**
